.github/actions/test/action.yml

```diff
@@ -3,8 +3,7 @@ description: "Run end-to-end tests. Note that this action assumes that Node and
 runs:
   using: "composite"
   steps:
-    - name: Install dependencies
-      run: npm install
+    - uses: bahmutov/npm-install@v1
```

Comment:
What do you think about using something a little more official like the GitHub action for node?  https://github.com/actions/setup-node#caching-global-packages-data